src/error.rs

```diff
@@ -289,6 +317,33 @@ impl de::Error for ConfigError {
     fn custom<T: fmt::Display>(msg: T) -> Self {
         Self::Message(msg.to_string())
     }
+
+    fn missing_field(field: &'static str) -> Self {
+        Self::NotFound(field.into())
+    }
```

Comment:
I'm willing to accept this now; the others will need to wait until a breaking change.

I would also prefer us to update the error message for this to better match serde's text.  I feel our own message is confusing.  Please change the message in a separate commit.